src/CLI/clitree/cli-xml/interface.xml

```diff
@@ -158,6 +244,279 @@ limitations under the License.
+        </COMMAND>
+		<COMMAND
+			name="switchport"
+			help="Configure switchport parameters"
+			>
+			<PARAM
+				name="switchport-subcommands"
+				help="Switchport subcommands"
+				ptype="SUBCOMMAND"
+				mode="switch"
+				>
+				<!--
+				<PARAM
+					name="mode"
+					help="Enter the port mode access/trunk"
+					ptype="SUBCOMMAND"
+					mode="subcommand"
+					>
+					<PARAM
+						name="mode subcommands"
+						help="Mode Subcommands"
+						ptype="SUBCOMMAND"
+						mode="switch"
+						>
+						<PARAM
+							name="access"
+							help="Port mode access"
+							ptype="SUBCOMMAND"
+							mode="subcommand"
+							>
+						</PARAM>
+					</PARAM>
+				</PARAM>
+				-->
+
+				<PARAM
+					name="access"
+					help="Set access mode characteristics of the interface"
+					ptype="SUBCOMMAND"
+					mode="subcommand"
+					>
+					<PARAM
+						name="vlan"
+						help="Set VLAN when interface is in access mode"
+						ptype="SUBCOMMAND"
+						mode="subcommand"
+						>
+						<PARAM
+							name="vlan-id"
+							help="VLAN ID of the VLAN when this port is in access mode"
+							ptype="UINT"
+							>
+						</PARAM>
+					</PARAM>
+				</PARAM>
+
+				<PARAM
+					name="trunk"
+					help="Configure trunking parameters on an interface"
+					ptype="SUBCOMMAND"
+					mode="subcommand"
+					>
+					<PARAM
+						name="allowed"
+						help="Configure trunking parameters on an interface"
+						ptype="SUBCOMMAND"
+						mode="subcommand"
+						>
+						<PARAM
+							name="vlan"
+							help="Configure trunking parameters on an interface"
+							ptype="SUBCOMMAND"
+							mode="subcommand"
+							>
+							<PARAM
+								name="vlan-id"
+                            	help="VLAN ID of the VLAN when this port is in access mode"
+                            	ptype="UINT"
+                            	>
+                        	</PARAM>
+						</PARAM>
+					</PARAM>
+				</PARAM>
+			</PARAM>
+                <ACTION>
+                     if test "${switchport-subcommands}" = "access"; then&#xA;
+                         python $SONIC_CLI_ROOT/sonic-cli-if.py patch_openconfig_vlan_interfaces_interface_ethernet_switched_vlan_config ${iface} ACCESS ${vlan-id}&#xA;
+                     else&#xA;
+                         python $SONIC_CLI_ROOT/sonic-cli-if.py patch_openconfig_vlan_interfaces_interface_ethernet_switched_vlan_config ${iface} TRUNK ${vlan-id}&#xA;
+                    fi
+
+                </ACTION>
+	</COMMAND>
+
+	<COMMAND
+		name="no switchport"
+		help="Remove switchport parameters"
+		>
+		<PARAM
+			name="switchport-subcommands"
+			help="Switchport subcommands"
+			ptype="SUBCOMMAND"
+			mode="switch"
+			>
+			<PARAM
+				name="access"
+				help="Remove access mode characteristics of the interface"
+				ptype="SUBCOMMAND"
+				mode="subcommand"
+				>
+				<PARAM
+					name="vlan"
+					help="Remove VLAN when interface is in access mode"
+					ptype="SUBCOMMAND"
+					mode="subcommand"
```

Comment:
Not sure why this PR has VLAN changes as well, it is possible to remove them and have only port-channel specific changes.